File: 백준/Gold/1931. 회의실 배정/회의실 배정.cc
#include <iostream>
#include <algorithm>

using namespace std;
pair<int, int> arr[100001];
int n, a, b;
int cnt = 1;

int main() {
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> b >> a;
        arr[i] = {a, b};
    }
    sort(arr, arr+n);
    int end = arr[0].first;
    for(int i = 1; i < n; i++) {
        if(end <= arr[i].second) {
            cnt++;
            end = arr[i].first;
        } 
    }
    cout << cnt;
}

File: 백준/Gold/1931. 회의실 배정/회의실 배정.kt
fun main() {
    var result = 1
    val meetingNum = readln().toInt()
    val meetings = List(meetingNum) {
        val (start, end) = readln()
            .split(" ")
            .map { it.toInt() }
        start to end
    }

    val sortedMeetings = meetings.sortedWith(
        compareBy <Pair<Int, Int>> { it.second }
            .thenBy { it.first }
    )

    var lastMeetingEndTime = sortedMeetings.first().second

    for (i in 1 until meetingNum) {
        if (lastMeetingEndTime <= sortedMeetings[i].first) {
            result++
            lastMeetingEndTime = sortedMeetings[i].second
        }
    }
    print(result)
}